indico/htdocs/js/indico/jquery/itempicker.js

```diff
@@ -0,0 +1,141 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    $.widget('indico.itempicker', {
+
+        options: {
+            onSelect: null,
+            onActionButtonClicked: null,
+            itemTemplate: null,
+            items: []
+        },
+
+        _create: function() {
+            var self = this;
+            var element = self.element;
+            var items = self.options.items.length ? self.options.items : (element.data('items') || []);
+            var dropdownContainer = $('<div>', {'class': 'dropdown-container'});
+            var filterWrapper = $('<div>', {'class': 'dropdown-filter-wrapper'});
+            var clearButtonWrapper = $('<span>', {'class': 'clear-button-wrapper'});
+            var filterInput = $('<input>', {'type': 'text', 'class': 'filter-input'});
+            var clearElement = $('<a>', {'class': 'icon-close'});
+            var itemsContainer = $('<div>', {'class': 'dropdown-items-container'});
+
+            clearElement.on('click', function() {
+                filterInput.val('').trigger('keyup');
+            });
+
+            clearButtonWrapper.append(clearElement);
+            filterWrapper.append(filterInput);
+            filterWrapper.append(clearButtonWrapper);
+            filterWrapper.on('keyup', function(evt) {
+                var $target = $(evt.target);
+                var textTyped = $target.val().trim();
+
+                if (!textTyped) {
+                    clearButtonWrapper.css('visibility', 'hidden');
+                } else {
+                    clearButtonWrapper.css('visibility', 'visible');
+                }
+
+                dropdownContainer.find('.dropdown-item').each(function() {
+                    var $item = $(this);
+                    if (!$item.data('filter').match(new RegExp(textTyped, 'i'))) {
+                        $item.hide();
```

Comment:
`toggle` introduces some untoward effects and that is why I used `hide` and `show`